eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ThrowFromClientLoggerCheck.java

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * To throw an exception,
+ * <ol>
+ *   <li>Must do it through a 'logger.logAndThrow', rather than by directly calling 'throw exception'</li>
+ *   <li>Always call return after calling 'logger.logAndThrow'.</li>
+ * </ol>
+ *
+ * Skip check if throwing exception from
+ * <ol>
+ *   <li>Static method</li>
+ *   <li>Static class</li>
+ *   <li>Constructor</li>
+ * </ol>
+ */
+public class ThrowFromClientLoggerCheck extends AbstractCheck {
```

Comment:
How does this check interact with the `JavadocThrowsCheck`? I would think this one breaks the other's logic as that was checking for LITERAL_THROW and those will be going away.

Reply:
@JonathanGiles  Should I need to check the JavaDocThrowsCheck or will be some other thoughts

Reply:
Alan's point is valid. The JavaDocThrowsCheck will need updating to understand the logAndThrow pattern to make sure we document these exceptions in JavaDoc.

Reply:
Created an issue to track this #4708